packages/astro/src/vite-plugin-astro-server/pipeline.ts

```diff
@@ -161,5 +177,54 @@ export class DevPipeline extends Pipeline {
 
 	clearRouteCache() {
 		this.routeCache.clearAll();
+		this.componentInterner = new WeakMap<RouteData, ComponentInstance>();
+	}
+
+	async getComponentByRoute(routeData: RouteData): Promise<ComponentInstance> {
+		const component = this.componentInterner.get(routeData);
+		if (component) {
+			return component;
+		} else {
+			const filePath = new URL(`./${routeData.component}`, this.config.root);
+			return await this.preload(routeData, filePath);
+		}
+	}
+
+	async tryRewrite(payload: RewritePayload): Promise<[RouteData, ComponentInstance]> {
+		let foundRoute;
+		if (!this.manifestData) {
+			throw new Error('Missing manifest data');
```

Comment:
```suggestion
			throw new Error('Missing manifest data. This is an internal error, please file an issue.');
```
I saw this recommendation to file an issue elsewhere. I like this pattern! If this is an internal error, it's worth calling out.